sample/CMakeLists.txt

```diff
@@ -1,9 +1,11 @@
+#include files 
 include_directories(${PROJECT_SOURCE_DIR}/commandlineparser/include)
 
-link_directories(${PROJECT_SOURCE_DIR}/commandlineparser/src)
-
+#executable for the sample app
 add_executable(SampleApp sample_user.cpp)
 
-target_link_libraries(SampleApp mylib)
+#linking library cmd_lib
+target_link_libraries(SampleApp cmd_lib)
 
+#install the targets 
 install(TARGETS SampleApp DESTINATION "${PROJECT_SOURCE_DIR}/install/bin")
```

Comment:
This is wrong install statement you are assuming your install directory 

install(TARGETS SampleApp 
  RUNTIME DESTINATION ${bin_dest} COMPONENT SampleApp 
  )